cm_ovirt_vm_creator.py

```diff
@@ -143,46 +143,83 @@ def construct_search_by_name_query(node_name):
     return search_string.format(node_name=node_name).__str__()
 
 
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(l), n):
+        yield l[i:i + n]
+
+
+def start_vm(vm_service, args):
+    """ Start a VM """
+    pub_sshkey = os.environ[args.pub_sshkey]
+    # ovirt API quirk - starting a VM with wait=False does nothing,
+    # and unlike other APIs, it accept async=True to make it asynchronous
+    vm_service.start(async=True,
+                     use_cloud_init=True,
+                     vm=types.Vm(initialization=types.Initialization(authorized_ssh_keys=pub_sshkey)))
+
+
 def create_vms(cluster_nodes, args):
     """ creates the vms in cluster_nodes list, and skipps if they exist """
     vm_dict = {}
-    for node in cluster_nodes:
-        print("node=%s" % (node), file=sys.stderr)
-        tmp = vms_service.list(search=construct_search_by_name_query(node))
-        if len(tmp) == 1:
-            vm_dict[node] = vms_service.vm_service(tmp[0].id)
-            print("VM %s was found ... skipping creation" % (node), file=sys.stderr)
-        else:
-            vm = vms_service.add(types.Vm(name=node,
-                                          cluster=types.Cluster(name=args.ovirt_cluster),
-                                          template=types.Template(name=args.ovirt_template)))
+    for block in chunks(cluster_nodes, args.block_size):
```

Comment:
I'd suggest to do a previous step to determine which virtual machines need to be created, outside of the "block" loop. Then, inside the "block" loop, you only need to create the virtual machines, with `wait=False`. After that you can wait for all the futures, and then start all the virtual machines that aren't started yet.